sdv/tabular/base.py

```diff
@@ -185,6 +187,62 @@ def sample(self, num_rows=None, max_retries=100):
 
         return sampled.head(num_rows)
 
+    def _make_conditions_df(self, conditions, num_rows):
+        if isinstance(conditions, pd.Series):
+            conditions = pd.DataFrame([conditions] * num_rows)
+
+        elif isinstance(conditions, dict):
+            try:
+                conditions = pd.DataFrame(conditions)
+            except ValueError:
+                conditions = pd.DataFrame([conditions] * num_rows)
+
+        elif not isinstance(conditions, pd.DataFrame):
+            raise TypeError("`conditions` must be a dataframe, a dictionary or a pandas series.")
+
+        return conditions
+
+    def sample(self, num_rows=None, max_retries=100, conditions=None):
+        """Write this."""
+        if conditions is None:
+            return self._sample_conditioned_rows(num_rows, max_retries, None)
+
+        # convert conditions to dataframe
+        conditions = self._make_conditions_df(conditions, num_rows)
+
+        # transform conditions, will use RDT when that's ready
+        all_conditions = []
+        for condition_column_name in conditions.columns:
+            condition_column = conditions[[condition_column_name]]
+            try:
+                transformed_conditions = self._metadata.transform(condition_column)
+                if len(transformed_conditions.columns) == 0:
+                    raise ValueError()
+                all_conditions.append(transformed_conditions)
+
+            except Exception:
+                raise ValueError(f'Cannot condition on {conditions}') from None
+
+        transformed_conditions = pd.concat(all_conditions, axis=1)
+        columns = transformed_conditions.columns
+        transformed_conditions["__condition_idx__"] = np.arange(len(transformed_conditions))
```

Comment:
Good solution! I was thinking about using a `reindex` to restore the order, but it seems like `reindex` would now work with duplicates, so I think this temporary indexes is the only option we have.